src/code/z_collision_check.c

```diff
@@ -1,392 +1,3373 @@
+    Collider_SetColliderBody(globalCtx, &dest->body, &src->body);
+    Collider_SetJntSphItemDim(globalCtx, &dest->dim, &src->dim);
+    return 1;
+}
+
+s32 Collider_JntSphItemSetAT(GlobalContext* globalCtx, ColliderJntSphItem* collider) {
+    Collider_ColliderBodySetAT(globalCtx, &collider->body);
+    return 1;
 }
 
-void func_8005BA1C(GlobalContext* globalCtx, ColliderBody* body) {
-    body->flags2 &= ~0x2;
+s32 Collider_JntSphItemSetAC(GlobalContext* globalCtx, ColliderJntSphItem* collider) {
+    Collider_ColliderBodySetAC(globalCtx, &collider->body);
+    return 1;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BA30.s")
+s32 Collider_JntSphItemSetOC(GlobalContext* globalCtx, ColliderJntSphItem* collider) {
+    Collider_ColliderBodySetOC(globalCtx, &collider->body);
+    return 1;
+}
 
-s32 func_8005BA74(UNK_TYPE arg0, UNK_TYPE arg1) {
+s32 Collider_InitJntSph(GlobalContext* globalCtx, ColliderJntSph* collider) {
+    Collider_InitBase(globalCtx, &collider->base);
+    collider->count = 0;
+    collider->list = NULL;
     return 1;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BA84.s")
+// Destruct ColliderJntSph (malloc)
+s32 Collider_FreeJntSph(GlobalContext* globalCtx, ColliderJntSph* collider) {
+    ColliderJntSphItem* next;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BAD8.s")
+    Collider_DestroyBase(globalCtx, &collider->base);
+    next = collider->list;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BB10.s")
+    while (next < collider->list + collider->count) {
+        Collider_DestroyJntSphItem(globalCtx, next);
+        next++;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BB48.s")
+    collider->count = 0;
+    if (collider->list != NULL) {
+        ZeldaArena_FreeDebug(collider->list, "../z_collision_check.c", 1393);
+    }
+    collider->list = NULL;
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BB8C.s")
+// Destruct ColliderJntSph (no malloc)
+s32 Collider_DestroyJntSph(GlobalContext* globalCtx, ColliderJntSph* collider) {
+    ColliderJntSphItem* next;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BBB0.s")
+    Collider_DestroyBase(globalCtx, &collider->base);
+    next = collider->list;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BBD4.s")
+    while (next < collider->list + collider->count) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BBF8.s")
+        Collider_DestroyJntSphItem(globalCtx, next);
+        next++;
+    }
+    collider->count = 0;
+    collider->list = NULL;
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BC28.s")
+// ClObjJntSph
+s32 Collider_SetJntSph_Set(GlobalContext* globalCtx, ColliderJntSph* dest, ColliderJntSphInit_Actor* src) {
```

Comment:
In general I think Joint Sphere written as JntSph can be confusing. I get alot of these function names would become incredibly long though. Just saying its a change to consider. If anything, at least a comment somewhere explaining what the JntSph abbreviation is and what it means